### PR TITLE
AQCU-1125: adding legend to derivation chain report including the pro…

### PR DESCRIPTION
**Pull request by mhines-usgs**
…cessor types

### CODE REVIEW

File: R/utils-render.R [hunk 1]
```diff
@@ -98,6 +98,9 @@ startTemplatedRender <- function(reportJson, author){
   #load header image
   templateData[["usgsLogoBase64"]] <- loadUsgsLogoBase64()
   
+  #load symbology processor images for derivation chain report
```
Comment (thongsav-usgs) [inline R/utils-render.R]: This should not be in the shared global function. There is a place for it in derivationchain-data.R.

File: R/utils-render.R [hunk 2]
```diff
@@ -194,6 +197,24 @@ loadUsgsLogoBase64 <- function() {
   return(base64encode(system.file('shared','usgs_logo.jpg', package = 'repgen')))
 }
 
+#' load legend symbology for derivation chain report
+#' @description will load the legend icons needed and convert to base64 text
+#' @return base64 string representation of legend images
+#' @importFrom base64enc base64encode
+loadLegendSymbologyDerivationChain <- function() {
```
Comment (thongsav-usgs) [inline R/utils-render.R]: I'd also move this function to derivationchain-data.R

File: inst/templates/derivationchain/body.mustache [hunk 3]
```diff
@@ -1,8 +1,20 @@
 <div id="cy"></div>
 <br/>
 <p><i>Time-Series processor icon images courtesy of Aquatic Informatics.</i></p>
+<div id="legendContainer"><strong>Legend</strong><br/><br/>
+<span style="text-decoration:underline">Processor Types:</span> <br/>
+<img src="data:image/jpeg;base64,{{derivationChainSymbols.ratingModelDerived}}" title="Rating derived" alt="Rating derived"> Rating derived<br/>
```
Comment (thongsav-usgs) [inline inst/templates/derivationchain/body.mustache]: Once you move the functions to derivationchain-data.R, you should be able to access the same information with reportData.derivationChainSymbols.ratingModelDerived

**Review by thongsav-usgs — CHANGES_REQUESTED** [review]

File: R/utils-render.R (hunk 2 above)
Comment (mhines-usgs) [reply R/utils-render.R]: :+1:

File: R/utils-render.R (hunk 1 above)
Comment (mhines-usgs) [reply R/utils-render.R]: duh, woops. :+1:

File: inst/templates/derivationchain/body.mustache (hunk 3 above)
Comment (mhines-usgs) [reply inst/templates/derivationchain/body.mustache]: hah, i missed this but you're correct :) :+1:

**Review by thongsav-usgs — APPROVED** [review]

**Pull request merged**